rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.vue

```diff
@@ -0,0 +1,161 @@
+                        {{ $t("file.warning.not.displayed.nonadmin.message") }}
+                      </template>
+                    </span>
+                  </template>
+                </div>
+              </div>
+            </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {defineComponent} from "vue";
+import AceEditorVue from "@/library/components/utils/AceEditorVue.vue";
+import AceEditor from "@/library/components/utils/AceEditor.vue";
+import {getRundeckContext} from "@/library";
+import {url} from  "@/library/rundeckService"
+import {Notification} from "uiv";
+
+const rundeckClient = getRundeckContext().rundeckClient
+export default defineComponent({
+  name: "EditProjectFile",
+  components: {AceEditor, AceEditorVue},
+  data() {
+    return {
+      fileText: '',
+      markdownSectionOpen: false,
+      errorMsg: ''
+    };
+  },
+  props: {
+    filename: '',
+    displayConfig: [],
+    project: '',
+    authAdmin: false,
+  },
+  computed: {
+    createProjectConfigureLink() {
+      return url('project/' + this.project + '/configure').href
+    },
+  },
+  mounted() {
+    this.getFileText()
+  },
+  methods: {
+    async saveProjectFile() {
+      if(this.fileText === ''){
+        this.fileText = '#This is a test'
+      }
+      const resp = await rundeckClient.sendRequest({
+        baseUrl: `${getRundeckContext().rdBase}api/${getRundeckContext().apiVersion}`,
+        pathTemplate: "/project/"+this.project+"/" + this.filename,
+        method: "PUT",
+        body: {
+          contents: this.fileText
+        }
+      });
+      if(resp.status === 200){
+        this.notifySuccess("Success", "Saved Project File " + this.filename )
+      }
+      else{
+        if(resp.status !== 200){
+          this.errorMsg = resp.parsedBody.message
+          this.notifyError(this.errorMsg)
+        }
+      }
+    },
+    createProjectHomeLink() {
+      document.location = url('project/' + this.project + '/home').href
+    },
+    notifyError(msg) {
+      Notification.notify({
+        type: "danger",
+        title: "An Error Occurred",
+        content: msg,
+        duration: 0
+      });
+    },
+
+    notifySuccess(title, msg) {
+      Notification.notify({
+        type: "success",
+        title: title,
+        content: msg,
+        duration: 5000
+      });
+    },
+    getFileText(){
+      rundeckClient.sendRequest({
+        baseUrl: `${getRundeckContext().rdBase}api/${getRundeckContext().apiVersion}`,
+        pathTemplate: "/project/"+this.project+"/" + this.filename,
+        headers: {
+          'Accept': 'application/json'
+        },
+        method: 'GET'
+      }).then(response => {
+        if(response.status === 200){
+          this.fileText = response.parsedBody.contents
+        }
+        else{
+          if(response.status !== 200){
+            this.errorMsg = response.parsedBody.message
+            this.notifyError(this.errorMsg)
+          }
```

Comment:
issue: When adding a message for the first time, this API is returning 404 as the file still doesn't exist. You might want to tweak this check to not output an error in this case.